pkg/hosts/hosts.go

```diff
@@ -169,7 +173,9 @@ func (h *Hosts) Remove(hosts []string) error {
 		for hostIdx, hostname := range line.Hostnames {
 			if _, ok := hostEntries[hostname]; ok {
 				if len(line.Hostnames) > 1 {
+					h.File.Lock()
```

Comment:
I'd take the lock around the whole loop (starting from `for i := start; i < end; i++ {`) and use `defer h.File.Unlock()` , but this won't go well with `RemoveHostsFileLineByRow` which also takes the lock :-/ Not really clear what the locking expectations are with libhosty.
It might be better to use our own lock, and to serialize all accesses to the public API. On Windows we access `pkg/hosts` through a REST API so our API will be called from multiple go routines.